src/DSP/FFT/RingBuffer.h

```diff
@@ -52,34 +68,61 @@ template<class T> class RingBuffer
     /// @param src The array of items to add to the buffer.
     /// @param count The amount of items to add to the buffer.
     /// @return The amount of items actually added to the buffer.
-    int Produce(const T* src, int32_t count)
+    std::size_t Produce(const T* src, std::size_t count)
     {
         std::lock_guard<std::mutex> lck(mMutex);
         count = std::min(count, capacity - size);
-        int produced = 0;
+        std::size_t produced = 0;
 
         if (tailIndex + count >= capacity)
         {
-            int toCopy = capacity - tailIndex;
-            memcpy(&buffer[tailIndex], src, toCopy * sizeof(T));
+            std::size_t toCopy = capacity - tailIndex;
+            for (std::size_t i = 0; i < toCopy; ++i)
+            {
+                buffer.at(i + tailIndex) = src[i];
+            }
+
             src += toCopy;
             count -= toCopy;
             size += toCopy;
             produced += toCopy;
             tailIndex = 0;
         }
-        memcpy(&buffer[tailIndex], src, count * sizeof(T));
+        for (std::size_t i = 0; i < count; ++i)
+        {
+            buffer.at(i + tailIndex) = src[i];
+        }
         tailIndex += count;
         size += count;
         produced += count;
         return produced;
     }
 
+    /// @brief Gets the amount of items in the buffer.
+    /// @return The amount of items currently in the buffer.
+    std::size_t Size() const { return size; }
```

Comment:
size, is neither atomic, nor guarded by lock, creates possibility for race condition.